README.md

```diff
@@ -1,8 +1,81 @@
-# sdg
+# Synthetic Data Generation (SDG)
 
 ![Lint](https://github.com/instructlab/sdg/actions/workflows/lint.yml/badge.svg?branch=main)
 ![Build](https://github.com/instructlab/sdg/actions/workflows/pypi.yaml/badge.svg?branch=main)
 ![Release](https://img.shields.io/github/v/release/instructlab/sdg)
 ![License](https://img.shields.io/github/license/instructlab/sdg)
 
 Python library for Synthetic Data Generation
+
+## Introduction
+
+Synthetic Data Generation (SDG) is a process that creates an artificially generated dataset that mimics real data based on provided examples. SDG uses a YAML file containing question-and-answer pairs as input data.
+
+## Installing the SDG library
+
+Clone the library and navigate to the repo:
+
+```bash
+git clone https://github.com/instructlab/sdg
+cd sdg
+```
+
+Install the library:
+
+```bash
+pip install .
+```
+
+## Using the library
+
+You can use the SDG library with the following items
+
+```bash
+ from instructlab.sdg.generate_data import generate_data
+ from instructlab.sdg.utils import GenerateException
+```
+
+<!--Not sure what more your thinking of adding here -->
+
+## Pipelines
+
+There are four pipelines that are used in SDG. Each pipeline requires specific hardware specifications.
+<!--TODO: Add explanations of pipelines-->
+
+*Full* -
+
+Minimum hardware requirements for running the simple pipeline
```

Comment:
might need someone else to keep me honest, but I'm not too sure it makes a lot of sense to specify "minimum" requirements here. The 'full' pipeline may run on smaller infra than we typically use, but the experience might be impacted (e.g. very slow, might crash, etc.).

Reply:
I think I was just thinking of a qualitative way to determine which pipeline to use. But more general might be better. I can say something like: This pipeline is targeted for running SDG on CPUs" or something?